web_src/js/components/DashboardRepoList.js

```diff
@@ -124,7 +124,7 @@ function initVueComponents() {
         return this.repos.length > 0 && this.repos.length < this.counts[`${this.reposFilter}:${this.archivedFilter}:${this.privateFilter}`];
       },
       searchURL() {
-        return `${this.subUrl}/api/v1/repos/search?sort=updated&order=desc&uid=${this.uid}&team_id=${this.teamId}&q=${this.searchQuery
+        return `${this.subUrl}/api/v1/repos/search?sort=updated&order=desc&minimal=1&uid=${this.uid}&team_id=${this.teamId}&q=${this.searchQuery
```

Comment:
It seems that `&q=${this.searchQuery}`, the value part should be escaped. Although the original code didn't do escape either, I think it's incorrect.

Reply:
Which use-case would be considered here?

Reply:
The idea `every parameter should be encoded correctly` can avoid bugs as much as possible. Why not encoding everything correctly .....

Reply:
> Why not encoding everything correctly .....

Because it's unnecessary, the characters that a repo can contain doesn't include any characters that would need to be escaped. There's no benefit of escaping them, the searching is done by a database query, so it doesn't allow `"test"` or any other string matching to work.

Reply:
Here is the demo:

If a user inputs "test&other", then the URL is:

`https://try.gitea.io/api/v1/repos/search?sort=updated&order=desc&uid=59078&team_id=0&q=test&other=&page=1&limit=15&mode=&archived=false`

The part `&q=test&other=&` is incorrect.

![image](https://user-images.githubusercontent.com/2114189/156367845-ab47ae19-a14e-4d87-8cf3-42a645215980.png)

Reply:
Escaping it or not it will have the same effect, no repo's will match the string. I'm not really sure why we're trying to handle a case which won't change the output anyway, I'm not _really_ finding this a bug that you can arbitrary add queries, as it wouldn't execute any javascript or be any different that someone replicating this via `curl`.

If repo's was able to have these kinds of characters as title or as topic it would be a different story, but not worth it to escape it IMO.

Reply:
Anyway, the old code just works, and there is no obvious security problem. It's just my personal opinion.

Indeed putting a string into a query parameter without encoding is not a good practice. Maybe I am too strict about the code style, personally I always do proper encoding for parameters.

Reply:
I don't mind putting it in if it's really necessary, but given it doesn't change the output I don't think we should consider escaping it.

Reply:
Nvm ... just personal opinion.